memberlist.go

```diff
@@ -579,9 +579,14 @@ func (m *Memberlist) Leave(timeout time.Duration) error {
 			return nil
 		}
 
+		// This dead message is special, because Node and From are the
+		// same. This helps other nodes figure out that a node left
+		// intentionally. When Node equals From, other nodes know for
+		// sure this node is gone.
 		d := dead{
 			Incarnation: state.Incarnation,
 			Node:        state.Name,
+			From:        state.Name,
```

Comment:
This doesn't look like a new field as I can't see it added in this PR, but why was it not being set at all before? Is it only used in some cases? Or is it populated during send and this addition is redundant?

Reply:
It was previously only set when one node suspected another one. Hasn't been set before when a node send this message while leaving.